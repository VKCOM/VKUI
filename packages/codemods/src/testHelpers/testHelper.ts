/**
 * Файл для упрощенной работы с дефолтными функциями запуска `jscodeshift`
 */
import path from 'path';
import {
  applyTransform,
  defineSnapshotTestFromFixture as defineSnapshotTestFromFixtureOrig,
  // @ts-expect-error: TS7016 no types for package
} from 'jscodeshift/dist/testUtils';
import { JSCodeShiftOptions } from '../types';

export { applyTransform };

export function getTestFixturesInputPath(
  dirName: string,
  testFilePrefix: string,
  extension = 'tsx',
) {
  const fixtureDir = path.join(dirName, '..', '__testfixtures__');
  return path.join(fixtureDir, testFilePrefix + `.input.${extension}`);
}

/**
 * У оригинального defineSnapshotTestFromFixture нет типов
 * Избавляемся от необходимости в тестах постоянно указывать `module`
 */
export function defineSnapshotTestFromFixture(
  dirName: string,
  transformName: string,
  options: JSCodeShiftOptions,
  testFilePrefix: string,
) {
  // Assumes transform is one level up from __tests__ directory
  const module = require(path.join(dirName, '..', transformName));
  defineSnapshotTestFromFixtureOrig(
    dirName,
    module,
    options,
    testFilePrefix,
    'transforms correctly',
    { parser: 'tsx' },
  );
}
