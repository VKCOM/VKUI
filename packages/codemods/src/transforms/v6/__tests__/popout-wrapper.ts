jest.autoMockOff();

import { defineSnapshotTestFromFixture } from '../../../testHelpers/testHelper';

const name = 'popout-wrapper';
const fixtures = ['basic'] as const;

describe(name, () => {
  fixtures.forEach((test) =>
    defineSnapshotTestFromFixture(__dirname, name, global.TRANSFORM_OPTIONS, `${name}/${test}`),
  );
});
