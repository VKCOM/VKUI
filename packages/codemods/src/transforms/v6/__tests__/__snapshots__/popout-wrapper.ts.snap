// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popout-wrapper transforms correctly 1`] = `
"import { PopoutWrapper } from '@vkontakte/vkui';
import React from 'react';

const App = () => {
  return (
    (<React.Fragment>
      <PopoutWrapper />
      <PopoutWrapper noBackground />
    </React.Fragment>)
  );
};"
`;
